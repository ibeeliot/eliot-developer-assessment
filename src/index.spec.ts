import * as request  from 'supertest';
import app from './index';
import { matchers } from 'jest-json-schema';
expect.extend(matchers);

interface Assessment {
	id: number,
	is_complete: boolean,
	start_time: string,
	total_time: string | null,
	progress: boolean | null,
	total_correct: number | null,
	build_seed: null,
	version_number: null,
	assessment: number,
	student: number,
	created_at: string,
	updated_at: string,
	created_by: null,
	updated_by: null,
	}


test('/ route returns simple hello', async () => {
	const response = await request(app.callback()).get('/');
	expect(response.status).toBe(200);
	expect(response.text).toBe('Hello World');
})


test('/student/display/:id returns normalized object', async () => {
	const expectedSchema: Assessment = ({
	id: 0,
	is_complete: false,
	start_time: '',
	total_time: '',
	progress: false,
	total_correct: null,
	build_seed: null,
	version_number: null,
	assessment: 0,
	student: 1,
	created_at: '',
	updated_at: '',
	created_by: null,
	updated_by: null,
	})

	const response = await request(app.callback()).get(`/student/display/${1}`);
	expect(response.status).toBe(200);
	expect(response.text).toMatchSnapshot();
	expect(response.text).toMatchSchema(expectedSchema);
})