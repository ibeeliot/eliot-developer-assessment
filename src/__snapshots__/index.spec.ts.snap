// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/student/display/:id returns normalized object 1`] = `"{\\"upcoming\\":[{\\"id\\":17,\\"is_complete\\":false,\\"start_time\\":\\"2021-10-06T15:42:57.000Z\\",\\"total_time\\":null,\\"progress\\":false,\\"total_correct\\":null,\\"build_seed\\":null,\\"version_number\\":null,\\"assessment\\":3,\\"student\\":1,\\"created_at\\":\\"2020-08-06T15:42:57.000Z\\",\\"updated_at\\":\\"2020-08-06T15:43:00.000Z\\",\\"created_by\\":null,\\"updated_by\\":null}],\\"open\\":[{\\"id\\":15,\\"is_complete\\":false,\\"start_time\\":\\"2020-08-06T15:42:57.000Z\\",\\"total_time\\":null,\\"progress\\":false,\\"total_correct\\":null,\\"build_seed\\":null,\\"version_number\\":null,\\"assessment\\":3,\\"student\\":1,\\"created_at\\":\\"2020-08-06T15:42:57.000Z\\",\\"updated_at\\":\\"2020-08-06T15:43:00.000Z\\",\\"created_by\\":null,\\"updated_by\\":null}],\\"expired\\":[{\\"id\\":19,\\"is_complete\\":false,\\"start_time\\":\\"2020-06-06T15:42:57.000Z\\",\\"total_time\\":\\"3325\\",\\"progress\\":false,\\"total_correct\\":5,\\"build_seed\\":null,\\"version_number\\":null,\\"assessment\\":7,\\"student\\":1,\\"created_at\\":\\"2020-08-06T15:42:57.000Z\\",\\"updated_at\\":\\"2020-08-06T15:43:00.000Z\\",\\"created_by\\":null,\\"updated_by\\":null}],\\"inProgress\\":[{\\"id\\":21,\\"is_complete\\":false,\\"start_time\\":\\"2020-08-08T15:42:57.000Z\\",\\"total_time\\":\\"11193\\",\\"progress\\":true,\\"total_correct\\":null,\\"build_seed\\":null,\\"version_number\\":null,\\"assessment\\":7,\\"student\\":1,\\"created_at\\":\\"2020-08-06T15:42:57.000Z\\",\\"updated_at\\":\\"2020-08-06T15:43:00.000Z\\",\\"created_by\\":null,\\"updated_by\\":null}],\\"completed\\":[{\\"id\\":7,\\"is_complete\\":true,\\"start_time\\":\\"2020-07-24T16:36:37.000Z\\",\\"total_time\\":\\"183297\\",\\"progress\\":false,\\"total_correct\\":5,\\"build_seed\\":null,\\"version_number\\":null,\\"assessment\\":1,\\"student\\":1,\\"created_at\\":\\"2020-07-24T16:36:37.000Z\\",\\"updated_at\\":\\"2020-07-24T16:39:41.000Z\\",\\"created_by\\":null,\\"updated_by\\":null},{\\"id\\":8,\\"is_complete\\":true,\\"start_time\\":\\"2020-07-24T16:49:59.000Z\\",\\"total_time\\":\\"267049636\\",\\"total_correct\\":2,\\"progress\\":false,\\"build_seed\\":null,\\"version_number\\":null,\\"assessment\\":2,\\"student\\":1,\\"created_at\\":\\"2020-07-24T16:49:59.000Z\\",\\"updated_at\\":\\"2020-07-27T19:00:49.000Z\\",\\"created_by\\":null,\\"updated_by\\":null},{\\"id\\":12,\\"is_complete\\":true,\\"start_time\\":\\"2020-07-27T20:41:55.000Z\\",\\"total_time\\":\\"6435786\\",\\"progress\\":false,\\"total_correct\\":5,\\"build_seed\\":null,\\"version_number\\":null,\\"assessment\\":4,\\"student\\":1,\\"created_at\\":\\"2020-07-27T20:41:55.000Z\\",\\"updated_at\\":\\"2020-07-27T22:29:11.000Z\\",\\"created_by\\":null,\\"updated_by\\":null}]}"`;
